includes/class-wc-youcanpay-api.php

```diff
@@ -4,247 +4,264 @@
-			$ip = $_SERVER['HTTP_CLIENT_IP'];
-		} elseif ( ! empty( $_SERVER['HTTP_X_FORWARDED_FOR'] ) ) {
-			//to check ip is pass from proxy
-			$ip = $_SERVER['HTTP_X_FORWARDED_FOR'];
-		} else {
-			$ip = $_SERVER['REMOTE_ADDR'];
-		}
-
-		return $ip;
-	}
+class WC_YouCanPay_API
+{
+
+    /**
+     * Secret API Key.
+     *
+     * @var string
+     */
+    private static $private_key = '';
+
+    /**
+     * Secret API Key.
+     *
+     * @var string
+     */
+    private static $public_key = '';
+
+    /**
+     * Sandbox Mode is enabled.
+     *
+     * @var string
+     */
+    private static $is_test_mode = '';
+
+    /**
+     * Set secret API Key.
+     *
+     * @param string $private_key
+     */
+    public static function set_private_key($private_key)
+    {
+        self::$private_key = $private_key;
+    }
+
+    /**
+     * Set secret API Key.
+     *
+     * @param string $public_key
+     */
+    public static function set_public_key($public_key)
+    {
+        self::$public_key = $public_key;
+    }
+
+    /**
+     * Set secret API Key.
+     *
+     * @param string $test_mode
+     */
+    public static function set_test_mode($test_mode)
+    {
+        self::$is_test_mode = ('yes' === $test_mode);
+    }
+
+    /**
+     * Get secret key.
+     *
+     * @return string
+     */
+    public static function get_private_key()
+    {
+        if (!self::$private_key) {
+            $options = get_option('woocommerce_youcanpay_settings');
+
+            if (isset($options['sandbox_mode'], $options['private_key'], $options['sandbox_private_key'])) {
+                self::set_private_key(
+                    'yes' === $options['sandbox_mode'] ? $options['sandbox_private_key'] : $options['private_key']
+                );
+            }
+        }
+
+        return self::$private_key;
+    }
+
+    /**
+     * Get public key.
+     *
+     * @return string
+     */
+    public static function get_public_key()
+    {
+        if (!self::$public_key) {
+            $options = get_option('woocommerce_youcanpay_settings');
+
+            if (isset($options['sandbox_mode'], $options['public_key'], $options['sandbox_public_key'])) {
+                self::set_public_key(
+                    'yes' === $options['sandbox_mode'] ? $options['sandbox_public_key'] : $options['public_key']
+                );
+            }
+        }
+
+        return self::$public_key;
+    }
+
+    /**
+     * Get public key.
+     *
+     * @return string
+     */
+    public static function is_test_mode()
+    {
+        if ('' === self::$is_test_mode) {
+            $options = get_option('woocommerce_youcanpay_settings');
+
+            if (isset($options['sandbox_mode'])) {
+                self::set_test_mode($options['sandbox_mode']);
+            }
+        }
+
+        return self::$is_test_mode;
+    }
+
+    /**
+     * Send the request to YouCan Pay's API
+     *
+     * @param WC_Order|WC_Order_Refund $order
+     * @param array $post_data
```

Comment:
the phpdoc should contain the exceptions also 🙏

Reply:
why? always we have only array as response

Reply:
https://github.com/NextmediaMa/youcan-pay-for-woocommerce/pull/12/commits/8f619601e2f3498862395a06dff350c1928dc177